pandas/core/indexes/multi.py

```diff
@@ -3218,6 +3218,8 @@ def equals(self, other) -> bool:
                 # other cannot contain tuples, so cannot match self
                 return False
 
+            if len(self) != len(other):
```

Comment:
can be an elif

Reply:
Implemented it, but is this necessary here? With the return statement, we will never reach that block if the first if condition was successful.
Wanted to be consistent with the lines above, because the same logic is used there with 

```
if self.is_(other):
    return True

if not isinstance(other, Index):
    return False

if not isinstance(other, MultiIndex):
    ...
```